### PR TITLE
Mosip 14065

### PR DESCRIPTION
**Pull request by urviljoshi**

### CODE REVIEW

File: hub/modules/connections/connections.bal [hunk 1]
```diff
@@ -57,3 +57,12 @@ public isolated function createMessageConsumer(websubhub:VerifiedSubscription me
...
     return check new (config:KAFKA_BOOTSTRAP_NODE, consumerConfiguration);  
 }
+
+// Consumer which reads the health check
+kafka:ConsumerConfiguration healthConsumerConfig = {
+    groupId: "health--check-topics-group-" + config:CONSTRUCTED_SERVER_ID,
+    offsetReset: "earliest",
+    topics: [ config:HEALTH_CHECK_TOPIC ]
+};
+public final kafka:Consumer healthCheckConsumer = check new (config:KAFKA_BOOTSTRAP_NODE, healthConsumerConfig);
```
Comment (vishwa-vyom) [inline hub/modules/connections/connections.bal:67]: Why are we creating this consumer, can we use one of the existing consumers?

File: hub/hub_service.bal [hunk 2]
```diff
@@ -21,18 +21,81 @@ import kafkaHub.security;
...
+        string consumerStatus="DOWN"; 
+        string kafkaStatus="DOWN";
+        kafka:TopicPartition[]|kafka:Error producerResult =  conn:statePersistProducer->getTopicPartitions(config:HEALTH_CHECK_TOPIC);
+        if(producerResult is kafka:TopicPartition[]){
+            producerStatus = "UP";
+        }
+        kafka:TopicPartition[]|kafka:Error result =  conn:healthCheckConsumer->getTopicPartitions(config:HEALTH_CHECK_TOPIC);
+        if(result is kafka:TopicPartition[]){
+            consumerStatus = "UP";
+        }
```
Comment (vishwa-vyom) [inline hub/hub_service.bal:51]: In both producer and consumer I see only getTopicPartitions being called, what is the point in having separate producer and consumer ?

**Review by vishwa-vyom — CHANGES_REQUESTED** [review]

File: hub/hub_service.bal (hunk 2 above)
Comment (urviljoshi) [reply hub/hub_service.bal:51]: using just producer to check connectivity.

File: hub/modules/connections/connections.bal (hunk 1 above)
Comment (urviljoshi) [reply hub/modules/connections/connections.bal:67]: using just producer to check connectivity.

Comment (vishwa-vyom) [reply hub/modules/connections/connections.bal:67]: Why is healthConsumerConfig section not removed ?

File: hub/hub_service.bal (hunk 2 above)
Comment (vishwa-vyom) [reply hub/hub_service.bal]: Resolved.

File: hub/modules/connections/connections.bal (hunk 1 above)
Comment (urviljoshi) [reply hub/modules/connections/connections.bal]: removed

**Review by vishwa-vyom — APPROVED** [review]

**Pull request merged**